org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/fix/Java50CleanUp.java

```diff
@@ -67,25 +70,33 @@ addAnotations && isEnabled(CleanUpConstants.ADD_MISSING_ANNOTATIONS_DEPRECATED)
 	protected ICleanUpFix createFix(CompilationUnit compilationUnit) throws CoreException {
 		boolean addAnotations= isEnabled(CleanUpConstants.ADD_MISSING_ANNOTATIONS);
 		boolean addOverride= isEnabled(CleanUpConstants.ADD_MISSING_ANNOTATIONS_OVERRIDE);
-		return Java50Fix.createCleanUp(compilationUnit,
+		ICleanUpFixCore cleanUpFixCore= Java50FixCore.createCleanUp(compilationUnit,
 				addAnotations && addOverride,
 				addAnotations && addOverride && isEnabled(CleanUpConstants.ADD_MISSING_ANNOTATIONS_OVERRIDE_FOR_INTERFACE_METHOD_IMPLEMENTATION),
 				addAnotations && isEnabled(CleanUpConstants.ADD_MISSING_ANNOTATIONS_DEPRECATED),
 				isEnabled(CleanUpConstants.VARIABLE_DECLARATION_USE_TYPE_ARGUMENTS_FOR_RAW_TYPE_REFERENCES));
+		return cleanUpFixCore == null ? null : new CleanUpFixWrapper(cleanUpFixCore);
 	}
 
 	@Override
 	protected ICleanUpFix createFix(CompilationUnit compilationUnit, IProblemLocation[] problems) throws CoreException {
 		if (compilationUnit == null)
 			return null;
 
+		ProblemLocationCore[] coreProblems= new ProblemLocationCore[problems.length];
+		for (int i= 0; i < coreProblems.length; i++) {
+			coreProblems[i]= new ProblemLocationCore(problems[i].getOffset(), problems[i].getLength(), problems[i].getProblemId(), problems[i].getProblemArguments(), problems[i].isError(),
+					problems[i].getMarkerType());
+		}
```

Comment:
Also at  https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/344ae32efe1d334d37635899ec2455a6aa665685/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/fix/CleanUpFixWrapper.java#L56-L59 . More common to go from IProblem -> IProblemLocationCore .

Is it worth adding an extra constructor to ProblemLocationCore that takes an IProblemLocation ?